compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.desktop.kt

```diff
@@ -203,9 +203,11 @@ internal class ComposeContainer(
         windowContext.setWindowFocused(isFocused)
         mediator.onChangeWindowFocus()
         layers.fastForEach(DesktopComposeSceneLayer::onChangeWindowFocus)
-        lifecycle.handleLifecycleEvent(
-            event = if (isFocused) Lifecycle.Event.ON_RESUME else Lifecycle.Event.ON_PAUSE
-        )
+        if (isFocused) {
+            lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
```

Comment:
As I mention in the other PR, we can't just call `lifecycle.handleLifecycleEvent`, because we can't guarantee the order of native events. There can be a race situation:
```
onChangeWindowFocus -> addNotify
```
The state will be invalid (should be RESUMED, but will be STARTED)

To solve this, we have to remember states (isFocused, isAdded, isDisposed, isIconified) and update lifecycle state depending on it (see the snippet)

Reply:
I see the concern, however, I would very much like to avoid duplicating state here and in Lifecycle object.
Let me investigate other options.

Reply:
I've added a check that should prevent the described behavior without the need to maintain full state in this class.

Reply:
We can:
1. receive `windowDeiconified` without `addNotify` => state will be RESUMED instead if STARTED
2. receive `addNotify` after we received focus and windows is maximized => state will be STARTED instead of RESUMED
3. receive `onChangeWindowFocus` without `addNotify`` => state will be RESUMED instead if STARTED

Reply:
I would still suggest to switch to changing state depending on other states instead of sending events. Otherwise we can come up to a complicated code handling all cases.